src/lb_data.c

```diff
@@ -1597,3 +1913,168 @@ int lb_io_read(lb_t * lb, int timestep, io_event_t * event) {
 
   return ifail;
 }
+
+/*****************************************************************************
+ *
+ * lb_graph_halo_send_create
+ *
+ *****************************************************************************/
+
+int lb_graph_halo_send_create(const lb_t * lb, lb_halo_t * h, int * send_count) {
+
+  assert(lb);
+  assert(h);
+
+  tdpAssert( tdpGraphCreate(&h->gsend.graph, 0) );
+
+  for (int ireq = 1; ireq < h->map.nvel; ireq++) {
+    tdpGraphNode_t kernelNode;
+    tdpKernelNodeParams kernelNodeParams = {0};
+    void * kernelArgs[3] = {(void *) &lb->target,
+                            (void *) &h->target,
+                            (void *) &ireq};
+    kernelNodeParams.func = (void *) lb_halo_enqueue_send_kernel;
+    //kernelNodeParams.func = (void *) lb_null_kernel;
+    dim3 nblk;
+    dim3 ntpb;
+    int scount = send_count[ireq]*lb_halo_size(h->slim[ireq]);
```

Comment:
I think we should completely eliminate graph kernel nodes (and mempcys) if the request size is zero. This may have unintended behavior.